### DISCUSSION TITLE
Content script registration "runtime" without specifying matches

### DISCUSSION THREAD

**Discussion by TomPradat** [wxt-dev/wxt#714] [General]
Right now it doesn't seem possible to register a content script using `defineContentScript` with a property "registration" equal to "runtime" and with no property "matches".

I believe it would make sense to be able to do that. If I understood correctly, when you have scripting permissions in your manifest, you don't need the host permission to inject the content script dynamically.

To throw in some code, here is what I would want to write :

```javascript
export default defineContentScript({
  registration: "runtime",
  async main(ctx) {}
});
```

**Comment by TomPradat** [comment]
After double-checking (see https://developer.chrome.com/docs/extensions/reference/api/scripting), even with `scripting` permission, you either need `host_permissions` or `activeTab` to be able to inject scripts dynamically.

But in the case where you have `activeTab` and `scripting` it would make sense not to have to define the "matches" property

**Comment by aklinker1** [comment]
I believe there's nothing stopping you from using an empty array. That, or you should use `defineUnlistedScript` instead of `defineContentScript`.

**Reply by TomPradat** [reply]
Indeed I missed the empty array option! Thank you.

The reason I'm not using `defineUnlistedScript` is that I want to inject UI into a Shadow DOM
